Source/NSPathControl.m

```diff
@@ -0,0 +1,396 @@
+      [self setURL: nil];
+      [self setDelegate: nil];
+      [self setAllowedTypes: [NSArray arrayWithObject: NSFilenamesPboardType]];
+    }
+  return self;
+}
+
+- (void) setPathStyle: (NSPathStyle)style
+{
+  [_cell setPathStyle: style];
+  [self setNeedsDisplay];
+}
+
+- (NSPathStyle) pathStyle
+{
+  return [_cell pathStyle];
+}
+
+- (NSPathComponentCell *) clickedPathComponentCell
+{
+  return [_cell clickedPathComponentCell];
+}
+
+- (NSArray *) pathComponentCells
+{
+  return [_cell pathComponentCells];
+}
+
+- (void) setPathComponentCells: (NSArray *)cells
+{
+  [_cell setPathComponentCells: cells];
+  [self setNeedsDisplay];
+}
+
+- (SEL) doubleAction;
+{
+  return [_cell doubleAction];
+}
+
+- (void) setDoubleAction: (SEL)doubleAction
+{
+  [_cell setDoubleAction: doubleAction];
+}
+
+- (NSURL *) URL
+{
+  return [_cell URL];
+}
+
+- (void) setURL: (NSURL *)url
+{
+  [_cell setURL: url];
+  [self setNeedsDisplay];
+}
+
+- (id<NSPathControlDelegate>) delegate
+{
+  return _delegate;
+}
+
+- (void) setDelegate: (id<NSPathControlDelegate>) delegate
+{
+  _delegate = delegate;
+}
+
+- (NSDragOperation) draggingSourceOperationMaskForLocal: (BOOL)flag
+{
+  if (flag)
+    {
+      return _localMask;
+    }
+
+  return _remoteMask;
+}
+
+- (void) setDraggingSourceOperationMask: (NSDragOperation)mask 
+                               forLocal: (BOOL)local
+{
+  if (local)
+    {
+      _localMask = mask;
+    }
+  else
+    {
+      _remoteMask = mask;
+    }
+}
+
+- (NSMenu *) menu
+{
+  return [super menu];
+}
+
+- (void) setMenu: (NSMenu *)menu
+{
+  [super setMenu: menu];
+}
+
+- (NSArray *) allowedTypes;
+{
+  return [_cell allowedTypes];
+}
+
+- (void) setAllowedTypes: (NSArray *)allowedTypes
+{
+  [_cell setAllowedTypes: allowedTypes];
+  [self registerForDraggedTypes: allowedTypes];
+}
+
+- (NSPathControlItem *) clickedPathItem
+{
+  return nil;
+}
+
+- (NSArray *) pathItems
+{
+  return _pathItems;
+}
+
+- (void) setPathItems: (NSArray *)items
+{
+  ASSIGNCOPY(_pathItems, items);
+  [self setNeedsDisplay];
+}
+
+- (NSAttributedString *) placeholderAttributedString
+{
+  return [_cell placeholderAttributedString];
+}
+
+- (void) setPlaceholderAttributedString: (NSAttributedString *)string
+{
+  [_cell setPlaceholderAttributedString: string];
+  [self setNeedsDisplay];
+}
+
+- (NSString *) placeholderString
+{
+  return [_cell placeholderString];
+}
+
+- (void) setPlaceholderString: (NSString *)string
+{
+  [_cell setPlaceholderString: string];
+  [self setNeedsDisplay];
+}
+
+- (NSColor *) backgroundColor
+{
+  return _backgroundColor;
+}
+
+- (void) setBackgroundColor: (NSColor *)color
+{
+  ASSIGN(_backgroundColor, color);
+  [self setNeedsDisplay];
+}
+
+- (void) setAction: (SEL)action
+{
+  _action = action;
+}
+
+- (SEL) action
+{
+  return _action;
+}
+
+- (void) setTarget: (id)target
+{
+  _target = target;
+}
+
+- (id) target
+{
+  return _target;
+}
+
+- (void) _doMenuAction: (id)sender
+{
+  if (_action)
```

Comment:
Is this the correct behaviour? I would have expected that we need to look at the menu item that causes this click.

Reply:
No, it actually returns the associated cell.  If you look at the test...

2020-05-05 16:01:02.469748-0400 NSPathControl_test[17687:4249934] clickedCell on dropdown <NSPathComponentCell: 0x600003300500> 'Macintosh SSD' _minWidth: 29.000000, _fullWidth: 104.000000, _resizedWidth: 0.000000, _currentWidth: 0.000000

It looks like it returns the cell.